src/lektorium/repo/local/objects.py

```diff
@@ -0,0 +1,72 @@
+import bidict
+import collections
+
+
+class Site(collections.abc.Mapping):
+    ATTR_MAPPING = bidict.bidict({
+        'name': 'site_name',
+        'email': 'custodian_email',
+        'owner': 'custodian',
+    })
+
+    def __init__(self, site_id, production_url, **props):
+        self.data = dict(props)
+        restricted_keys = ('sessions', 'staging_url')
+        if set(self.data.keys()).intersection(restricted_keys):
+            raise RuntimeError()
+        self.data['site_id'] = site_id
+        self.data['staging_url'] = None
+        self.production_url = production_url
+        self.sessions = {}
+
+    def __getitem__(self, key):
+        if key == 'sessions':
+            return list(self.sessions.values())
+        elif key == 'production_url':
+            result = self.production_url
+            if callable(self.production_url):
+                self.production_url, result = result()
```

Comment:
Seems strange that `callable` is checking `production_url` but a call is issued on `result`